### PR TITLE
Send endpoint to S3Uploader

### PR DESCRIPTION
**Pull request by BoPeng**
Currently the `S3Uploader` `fetch` from a relative URL `s3upload`

https://github.com/mbraak/django-file-form/blob/6fb9e11573ab45806c29b5716f2b7d84783efc2a/frontend/src/s3_uploader.ts#L63

which does not work if the form is not at the root.

### CODE REVIEW

File: django_file_form/forms.py [hunk 1]
```diff
@@ -18,17 +18,16 @@ def __init__(self, *args, **kwargs):
...
+            self.add_hidden_field('upload_url', reverse('tus_upload'))
 
         self.add_hidden_field('form_id', uuid.uuid4())
-        self.add_hidden_field('upload_url', self.get_upload_url())
         self.add_placeholder_inputs()
 
     def add_hidden_field(self, name, initial):
         self.fields[name] = CharField(widget=HiddenInput, initial=initial, required=False)
 
-    def get_upload_url(self):
```
Comment (BoPeng) [inline django_file_form/forms.py:29]: Removed because it is short but should restore if `get_upload_url` is an interface that needs to be exposed.

**Comment by mbraak** [comment]
I will have a look!

File: frontend/src/s3_uploader.ts [hunk 2]
```diff
@@ -60,7 +65,7 @@ const createMultipartUpload = (
     "content-type": "application/json",
     "X-CSRFToken": csrftoken
   });
-  return fetch("s3upload/", {
+  return fetch(endpoint, {
```
Comment (mbraak) [inline frontend/src/s3_uploader.ts:68]: This resulted in url ending with '//'. Fixed by removing the '/'

File: frontend/src/s3_uploader.ts [hunk 3]
```diff
@@ -79,13 +84,14 @@ const createMultipartUpload = (
...
-  uploadId
+  uploadId,
+  endpoint
 }: MultipartUpload): Promise<ServerPart[]> => {
   const filename = encodeURIComponent(key);
   const uploadIdEnc = encodeURIComponent(uploadId);
-  return fetch(`s3upload/${uploadIdEnc}?key=${filename}`, {
-    method: "get"
-  })
+  const url = urljoin(endpoint, uploadIdEnc, `?key=${filename}`);
```
Comment (mbraak) [inline frontend/src/s3_uploader.ts:92]: Use `urljoin` to avoid duplicate '/'

**Comment by mbraak** [comment]
I got an error that the url `/s3upload//` was invalid. I added a commit to this pr to fix that.

**Comment by BoPeng** [comment]
Sorry. The initial version has an endpoint without trailing slash but the new one has but I forgot to remove duplicated slashes.

**Comment by BoPeng** [comment]
Just tested and confirmed that the PR works fine. I think it is ready to be merged.

**Pull request merged**